installers/msi-language/StateDeploy/CustomAction.cs

```diff
@@ -250,21 +250,44 @@ private static ActionResult _installStateTool(Session session, out string stateT
             if (oldPath.Contains(stateToolInstallDir))
             {
                 session.Log("State tool installation already on PATH");
-                return ActionResult.Success;
             }
-
-            var newPath = string.Format("{0};{1}", stateToolInstallDir, oldPath);
-            session.Log(string.Format("updating PATH to {0}", newPath));
-            try
+            else
             {
-                Environment.SetEnvironmentVariable("PATH", newPath, EnvironmentVariableTarget.Machine);
+                var newPath = string.Format("{0};{1}", stateToolInstallDir, oldPath);
+                session.Log(string.Format("updating PATH to {0}", newPath));
+                try
+                {
+                    Environment.SetEnvironmentVariable("PATH", newPath, EnvironmentVariableTarget.Machine);
+                }
+                catch (Exception e)
+                {
+                    string msg = string.Format("Could not update PATH. Encountered exception: {0}", e.Message);
+                    session.Log(msg);
+                    new SecurityError().SetDetails(session, msg);
+                    return ActionResult.Failure;
+                }
             }
-            catch (Exception e)
+
+            session.Log("Running prepare step...");
+            string prepareCmd = " --output=json" + " _prepare";
```

Comment:
Nit: lead with the command, not the flags.
```suggestion
            string prepareCmd = " _prepare --output=json";
```